File: nvxapp.server.service/ClientServer-Service/GestionePresenze/Az_SubCommessaAttivitaService/Models/Az_SubCommessaAttivitaModel.cs
using nvxapp.server.service.ClientServer_Service.ModelsBase;

namespace nvxapp.server.service.ClientServer_Service.GestionePresenze.Az_SubCommessaAttivitaService.Models
{
    public class Az_SubCommessaAttivitaModel
    {
        public int Id { get; set; }
        public int IdAz_SubCommessa { get; set; }
        public int IdPar_Attivita { get; set; }
        public Boolean Default { get; set; }
    }



    public class Az_SubCommessaAttivita_GetAll_InModel { }

    public class Az_SubCommessaAttivita_GetAll_OutModel : ModelResult
    {
        public List<Az_SubCommessaAttivitaModel> Az_SubCommessaAttivita { get; set; } = new List<Az_SubCommessaAttivitaModel>();
    }

    public class Az_SubCommessaAttivita4EditModel : Az_SubCommessaAttivitaModel, ICheckObj<int>
    {
        public bool Checked { get; set; }
    }

    public class Az_SubCommessaAttivita_Get4SubCommessa_InModel
    {
        public int IdAz_SubCommessa { get; set; }
    }

    public class Az_SubCommessaAttivita_Get4SubCommessa_OutModel : ModelResult
    {
        public List<Az_SubCommessaAttivita4EditModel> Az_SubCommessaAttivita { get; set; } = new List<Az_SubCommessaAttivita4EditModel>();
    }

    public class Az_SubCommessaAttivita_Put4SubCommessa_InModel
    {
        public int IdAz_SubCommessa { get; set; }
        public List<Az_SubCommessaAttivita4EditModel> Az_SubCommessaAttivita { get; set; } = new List<Az_SubCommessaAttivita4EditModel>();
    }

    public class Az_SubCommessaAttivita_Put4SubCommessa_OutModel : ModelResult
    {
        public List<Az_SubCommessaAttivita4EditModel> Az_SubCommessaAttivita { get; set; } = new List<Az_SubCommessaAttivita4EditModel>();
    }



    public class Az_SubCommessaAttivita_4FullListModel
    {
        public int Commessa_Id { get; set; }
        public int Commessa_IdAz_Cliente { get; set; }
        public string Commessa_Decrizione { get; set; } = string.Empty;
        public Boolean Commessa_Default { get; set; }

        public int SubCommessa_Id { get; set; }
        public string SubCommessa_Decrizione { get; set; } = string.Empty;
        public Boolean SubCommessa_Default { get; set; }

        public int SubCommessaAttivita_Id { get; set; }
        public string SubCommessaAttivita_Decrizione { get; set; } = string.Empty;
        public Boolean SubCommessaAttivita_Default { get; set; }
        public int SubCommessaAttivita_IdPar_Attivita { get; set; }
    }

    public class Az_SubCommessaAttivita_GetAll_4FullList_InModel { }

    public class Az_SubCommessaAttivita_GetAll_4FullList_OutModel : ModelResult
    {
        public List<Az_SubCommessaAttivita_4FullListModel> Az_SubCommessaAttivita { get; set; } = new List<Az_SubCommessaAttivita_4FullListModel>();
    }


}

File: nvxapp.client/src/nvx/ClientServer-Service/GestionePresenze/Az_SubCommessaAttivita/Models/az-subcommessa-attivita-model.ts
import { ModelResult } from "../../../ModelsBase/model-result";
import { ICheckObj } from "../../../ModelsBase/check-obj";

export class Az_SubCommessaAttivitaModel {
  id: number = 0;
  idAz_SubCommessa: number = 0;
  idPar_Attivita: number = 0;
  default: boolean = false;
}

export class Az_SubCommessaAttivita4EditModel extends Az_SubCommessaAttivitaModel implements ICheckObj<number> {
  checked: boolean = false;
}

export class Az_SubCommessaAttivita_GetAll_InModel {}

export class Az_SubCommessaAttivita_GetAll_OutModel extends ModelResult {
  az_SubCommessaAttivita: Az_SubCommessaAttivitaModel[] = [];
}

export class Az_SubCommessaAttivita_Get4SubCommessa_InModel {
  idAz_SubCommessa: number = 0;
}

export class Az_SubCommessaAttivita_Get4SubCommessa_OutModel extends ModelResult {
  az_SubCommessaAttivita: Az_SubCommessaAttivita4EditModel[] = [];
}

export class Az_SubCommessaAttivita_Put4SubCommessa_InModel {
  idAz_SubCommessa: number = 0;
  az_SubCommessaAttivita: Az_SubCommessaAttivita4EditModel[] = [];
}

export class Az_SubCommessaAttivita_Put4SubCommessa_OutModel extends ModelResult {
  az_SubCommessaAttivita: Az_SubCommessaAttivita4EditModel[] = [];
}

export class Az_SubCommessaAttivita_4FullListModel {
  commessa_Id: number = 0;
  commessa_IdAz_Cliente: number = 0;
  commessa_Decrizione: string = '';
  commessa_Default: boolean = false;

  subCommessa_Id: number = 0;
  subCommessa_Decrizione: string = '';
  subCommessa_Default: boolean = false;

  subCommessaAttivita_Id: number = 0;
  subCommessaAttivita_Decrizione: string = '';
  subCommessaAttivita_Default: boolean = false;
  subCommessaAttivita_IdPar_Attivita: number = 0;
}

export class Az_SubCommessaAttivita_GetAll_4FullList_InModel {}

export class Az_SubCommessaAttivita_GetAll_4FullList_OutModel extends ModelResult {
  az_SubCommessaAttivita: Az_SubCommessaAttivita_4FullListModel[] = [];
}
